waspc/src/Analyzer/Evaluator/Combinators.hs

```diff
@@ -0,0 +1,176 @@
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | This module contains combinators for building evaluators for converting
+-- "TypedExpr" to arbitrary types.
```

Comment:
`"TypedExpr" to arbitrary types` -> it is still hard for me to understand what this means, it requires me having a lot of context to understand this piece of sentence.
Maybe we could make it more verbose. Smth like

> `TypeChecker.AST` declarations into their corresponding instances of `IsDeclType`. So basically from Wasp types into Haskell types.

I hope I said this correctly?